shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/IntervalShardingAlgorithm.java

```diff
@@ -143,8 +143,20 @@ public String doSharding(final Collection<String> availableTargetNames, final Pr
     }
     
     private boolean hasIntersection(final Range<LocalDateTime> calculateRange, final Range<Comparable<?>> range) {
-        LocalDateTime lower = range.hasLowerBound() ? parseDateTime(range.lowerEndpoint().toString()) : dateTimeLower;
-        LocalDateTime upper = range.hasUpperBound() ? parseDateTime(range.upperEndpoint().toString()) : dateTimeUpper;
+        LocalDateTime lower;
+        if (range.hasLowerBound()) {
+            Comparable<?> lowerEndpoint = range.lowerEndpoint();
+            lower = lowerEndpoint instanceof LocalDateTime ? parseDateTime(((LocalDateTime) lowerEndpoint).format(dateTimeFormatter)) : parseDateTime(lowerEndpoint.toString());
+        } else {
+            lower = dateTimeLower;
+        }
+        LocalDateTime upper;
+        if (range.hasUpperBound()) {
+            Comparable<?> upperEndpoint = range.upperEndpoint();
+            upper = upperEndpoint instanceof LocalDateTime ? parseDateTime(((LocalDateTime) upperEndpoint).format(dateTimeFormatter)) : parseDateTime(upperEndpoint.toString());
```

Comment:
@linghengqian Thank for your pull request, but i think it should consider `Date` type

Reply:
Maybe solved. There doesn't seem to be a `Util` class that lets me use pattern matching like JDK 17.